Gatsby UX Capture plugin and example

sergeychernyshev opened the pull request:
Gatsby integration plugin:
- injects inline copy of UXCapture library
- fires `UXCapture.startTransition()` before routes change

Example Gatsby site with UX Capture integration:
- injects library initialization using `<UXCaptureCreate />`
- injects marks into Gatsby-generated static HTML files
- works in both `page view` and `interactive view` modes (has 3 sample views)
- does NOT include mark and measure visualizations

Pivotal ticket: https://www.pivotaltracker.com/story/show/169297083

sergeychernyshev committed:
Added first version of Gatsby UX Capture example package

sergeychernyshev committed:
Added key to ux capture script

sergeychernyshev committed:
Removed all metric log functionality, just keeping core implementation. Works in page view mode.

sergeychernyshev committed:
Using onPreRouteUpdate to fire startTransition()

sergeychernyshev committed:
Remove duplicate UXCapture.create() call (was done using <UXCaptureCreate/>)

sergeychernyshev committed:
Moved gatsby configuration into a standalone plugin

sergeychernyshev committed:
Converted mark strings to constants, made strings unique.

sergeychernyshev committed:
Readme for UX Capture Gatsby Example

sergeychernyshev committed:
Gatsby plugin README

sergeychernyshev committed:
Public package

sergeychernyshev committed:
Publish

 - @meetup/gatsby-plugin-ux-capture@0.1.0
 - gatsby-ux-capture-example@0.1.1
 - @meetup/react-ux-capture-example@1.2.3
 - @meetup/react-ux-capture@1.3.2
 - @meetup/ux-capture@4.3.1

sergeychernyshev committed:
Namespaced Gatsby example

sergeychernyshev committed:
Publish

 - @meetup/gatsby-ux-capture-example@0.1.2

sergeychernyshev committed:
Updated license and removed example project CHANGELOG

sergeychernyshev committed:
Publish

 - @meetup/gatsby-plugin-ux-capture@0.1.1
 - @meetup/gatsby-ux-capture-example@0.1.3

sergeychernyshev committed:
Upgraded node in Travis

sergeychernyshev committed:
Moved create back to UXCaptureCreate component

cranberyxl commented on packages/gatsby-plugin-ux-capture/.gitignore:
```diff
@@ -0,0 +1,6 @@
+/*.js
+!index.js
+yarn.lock
```
Why ignore the yarn.lock?

sergeychernyshev replied:
Probably a default from plugin repo I copied. In this case though, yarn.lock is in the root of the monorepo anyway.

sergeychernyshev replied:
I'll remove it.

sergeychernyshev committed:
Don't ignore yarn.lock (real one is in the monorepo root anyway)

sergeychernyshev commented:
Hmm, looks like some issues with onMark handler in the code - didn't see it before... checking.

I'll probably refactor it a tiny bit as I did in nextjs PR to remove some extraneous abstractions.

sergeychernyshev commented:
Ignore my comment about `onMark` - it's TypeScript warnings.
Probably a separate ticket to define TypeScript annotations for the whole UX Capture.

sergeychernyshev committed:
Added "dev" target to unify with nextjs and remove extraneous View wrapper

sergeychernyshev commented:
Made some minor adjustments, but for some reason getting oMark errors when I run `gatsby developer` and no errors if I run `gatsby build && gatsby serve`.

sergeychernyshev committed:
Fixed UX Capture core library to make sure onMark and onMeasure stubs are set by default

sergeychernyshev committed:
Publish

 - @meetup/gatsby-plugin-ux-capture@0.1.2
 - @meetup/gatsby-ux-capture-example@0.1.4
 - @meetup/react-ux-capture-example@1.2.4
 - @meetup/react-ux-capture@1.3.3
 - @meetup/ux-capture@4.3.2
